xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearch.xml

```diff
@@ -198,14 +198,33 @@
     }
   };
 
-  var addFacetValueCheckbox = function() {
-    var checkBox = $(document.createElement('input')).attr('type', 'checkbox');
+  var addFacetValueCheckbox = function(index) {
+    // Create an id unique to the facet value.
+    let facetContainer = $(this).parents('.search-facet').first();
+    let checkBoxId = facetContainer.attr('data-name') + '-' + index.toString() + '-checkbox';
```

Comment:
What about checking if an element with the id exists already, just to be sure?

Reply:
Note: If the facetcontainer data-name is not unique, the setSearchUIState bugs out anyways.

Reply:
The `index` part is unique to the group of checkboxes in the same container (generated by the jquery `each` on [L252](https://github.com/xwiki/xwiki-platform/blob/f86702aba120ce73d099b8dd3385d1d293084e45/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearch.xml#L252)).

So the id generated should be unique anyways.